scripts/run_backend_tests.py

```diff
@@ -285,11 +285,15 @@ def _check_shards_match_tests(include_load_tests=True):
         return ''
     if len(set(shard_modules)) != len(shard_modules):
         # A module is duplicated, so we find the duplicate.
-        for module in shard_modules:
+        # We add no-cover because the case when the duplicate is not
+        # found will not arise. All elements in a set are unique and when
+        # len(set(shard_modules)) != len(shard_modules), there has to be
+        # atleast one duplicate module in shard_modules.
```

Comment:
Yes, I agree. By the above comment I mean that **If** the condition that `len(set(shard_modules)) != len(shard_modules)` is true, there would definitely be a duplicate in shard_modules. The branch 292-296 (exiting the for loop) would never happen since we will always catch a duplicate midway.

Reply:
OK, in that case, can you clarify in the comment that the no-cover here is for the branch condition where the loop terminates instead of being exited early via `break`?

Reply:
Done